projects/storefrontlib/cms-components/myaccount/new-consent-management/components/new-consent-management.component.ts

```diff
@@ -0,0 +1,340 @@
+/*
+ * SPDX-FileCopyrightText: 2023 SAP Spartacus team <spartacus-team@sap.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { Component, OnDestroy, OnInit } from '@angular/core';
+import {
+  AnonymousConsentsConfig,
+  AnonymousConsentsService,
+  AuthService,
+  ConsentTemplate,
+  GlobalMessageService,
+  GlobalMessageType,
+  UserConsentService,
+} from '@spartacus/core';
+import {
+  BehaviorSubject,
+  combineLatest,
+  concat,
+  Observable,
+  Subscription,
+} from 'rxjs';
+import {
+  distinctUntilChanged,
+  filter,
+  map,
+  scan,
+  skipWhile,
+  tap,
+  withLatestFrom,
+} from 'rxjs/operators';
+import { ConsentManagementComponentService } from '../../consent-management/consent-management-component.service';
+
+@Component({
+  selector: 'cx-new-consent-management',
+  templateUrl: './new-consent-management.component.html',
+})
+export class NewConsentManagementComponent implements OnInit, OnDestroy {
+  private subscriptions = new Subscription();
```

Comment:
Please use `protected` instead of `private`.

Reply:
modified as suggestion